qcodes_contrib_drivers/drivers/Tektronix/Keithley_6500.py

```diff
@@ -2,55 +2,154 @@
+        Args:
+            name: Name of instrument
+            address: Address of instrument
+            terminator: Termination character for SCPI commands
+            **kwargs: Keyword arguments to pass to __init__ function of VisaInstrument class
+        """
         super().__init__(name, address, terminator=terminator, **kwargs)
+        for quantity in ['VOLT', 'CURR', 'RES', 'FRES']:
+            channel = Keithley_Sense(self, quantity.lower(), quantity)
+            self.add_submodule(quantity.lower(), channel)
+
+        self.add_parameter('active_terminal',
+                           label='active terminal',
+                           get_cmd="ROUTe:TERMinals?",
+                           docstring="Active terminal of instrument. Can only be switched via knob on front panel.")
+
+        self.add_parameter('resistance',
+                           unit='Ohm',
+                           label='Measured resistance',
+                           get_parser=float,
+                           get_cmd=partial(self.measure, 'RES'),
+                           )
 
-        for sense in ['VOLT', 'CURR', 'RES', 'FRES']:
-            channel = Keithley_Sense(self, sense.lower(), sense)
-            self.add_submodule(sense.lower(), channel)
+        self.add_parameter('resistance_4w',
+                           unit='Ohm',
+                           label='Measured resistance',
+                           get_parser=float,
+                           get_cmd=partial(self.measure, 'FRES')
+                           )
+
+        self.add_parameter('voltage_dc',
+                           unit='V',
+                           label='Measured DC voltage',
+                           get_parser=float,
+                           get_cmd=partial(self.measure, 'VOLT')
+                           )
+
+        self.add_parameter('current_dc',
+                           unit='A',
+                           label='Measured DC current',
+                           get_parser=float,
+                           get_cmd=partial(self.measure, 'CURR')
+                           )
 
         self.connect_message()
+
+        # check if scanner card is connected
+        # If no scanner card is connected, the query below returns "Empty Slot".
+        # For the Scanner Card 2000-SCAN used for development of this driver the output was
+        # "2000,10-Chan Mux,0.0.0a,00000000".
+        scan_idn_msg = self.ask(":SYSTem:CARD1:IDN?")
+        if scan_idn_msg != "Empty Slot":
+            msg_parts = scan_idn_msg.split(",")
+            print(f"Scanner card {msg_parts[0]}-SCAN detected.")
+            for ch_number in range(1, 11):
+                scan_channel = Keithley_2000_Scan_Channel(self, ch_number)
+                self.add_submodule(f"ch{ch_number:d}", scan_channel)
+
+    # only measure if front terminal is active
+    def measure(self, quantity: str) -> str:
```

Comment:
As far as I can see the user should never call this directly so perhaps it should be called `_measure` to indicate that is